core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -95,6 +95,9 @@ class DiplomacyManager() {
     /** Total of each turn Science during Research Agreement */
     var totalOfScienceDuringRA = 0
 
+    /** Action to take when a unit grant should be made */
+    @Transient var onUnitGrant: () -> Unit = { civInfo.giftMilitaryUnitTo(otherCiv()) }
```

Comment:
There is no reason to make this an action. If you want to test this, you can create a small map with two small cities, and then check that the recieving civ has another unit.
This will also allow you to test what happens if one of the civs doesn't gave a city ;)

Reply:
Is there an example somewhere in the unit tests of creating a map like this?

Reply:
Yes, I think in the movement tests, where we check if a unit can enter enemy territory